src/main/java/net/minecraftforge/network/ServerStatusPing.java

```diff
@@ -58,21 +66,50 @@
  *          }
  *     ]
  * }
+ * </pre>
+ * </code>
+ * <p>
+ * Due to size of the ping packet (32767 UTF-16 code points of JSON data) this could exceed this limit and
+ * cause issues. To work around this, a truncation mechanism was introduced, to heuristically truncate the size of the
+ * data, at the expense of making the compatibility info on the server screen inaccurate.
+ *
+ * <p>
+ * Modern versions will send binary data, which is encoded in a custom format optimized for UTF-16 code point count.
+ * See {@link Serializer#encodeOptimized(ByteBuf)} and {@link Serializer#decodeOptimized(String)}.
+ * Essentially 15 bits of binary data are encoded into every UTF-16 code point. The resulting string is then stored in
+ * the "d" property of the resulting JSON.
+ *
+ * <p>
+ * For the format of the binary data see {@link Serializer#serialize(ServerStatusPing)}.
+ *
+ * <p>
+ * The "channels" and "mods" properties are retained for backwards compatibility,
+ * but left empty. A client that cannot read the old format would not be able to connect anyways, but the properties
+ * must exist to not cause exceptions.
+ *
+ * <code>
+ * <pre>
+ * {
+ *     "fmlNetworkVersion": FMLNETVERSION,
+ *     "channels": [],
+ *     "mods": [],
+ *     "d": "&lt;binary data&gt;"
+ * }
+ * </pre>
+ * </code>
```

Comment:
This should use a `{@code}` block tag wrapped in `<pre>`:
```suggestion
 * <pre>{@code
 * {
 *     "fmlNetworkVersion": FMLNETVERSION,
 *     "channels": [],
 *     "mods": [],
 *     "d": "&lt;binary data&gt;"
 * }
 * }</pre>
```